### PR TITLE
Add a `flutter create --list-samples` command

### LINKED ISSUE #28896: Error creating project with sample content
_From @akireyev on February 27, 2019 18:29_

## Steps to Reproduce

* Run a 'New Flutter Project' wizard in AS.
* Select 'generate sample content' with 'RaisedButton'

Expected: a new project is created.
Actual: an error in the log
```
Error creating project
Download failed.
URL: https://master-docs-flutter-io.firebaseapp.com/snippets/material.RaisedButton.dart
Error: 404 Not Found
```

## Version info

[√] Flutter (Channel beta, v1.1.8, on Microsoft Windows [Version 10.0.17134.619], locale en-US)
    • Flutter version 1.1.8 at ...
    • Framework revision 985ccb6d14 (7 weeks ago), 2019-01-08 13:45:55 -0800
    • Engine revision 7112b72cc2
    • Dart version 2.1.1 (build 2.1.1-dev.0.1 ec86471ccc)

[√] Android Studio (version 3.3)
    • Android Studio at D:\code\tools\JetBrains Toolbox\apps\AndroidStudio\ch-0\182.5264788
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

_Copied from original issue: flutter/flutter-intellij#3236_

### PR DESCRIPTION
**Pull request by DanTup**
## Description

Adds an option to `flutter create` to list the available samples. This is to be used by editors to get a list of valid samples without having to fetch them from the server themselves (which would require duplicating the logic for what the URL/hostname in the editors).

Called like this:

```
$ flutter create --list-samples samples.json
```

@pq @gspencergoog 

## Related Issues

- Fixes #28896.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by pq — APPROVED** [review]
👍

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -95,6 +95,12 @@ class CreateCommand extends FlutterCommand {
       defaultsTo: null,
       valueHelp: 'id',
     );
+    argParser.addOption(
+      'list-samples',
+      help: 'Specifies a file to output a list of Flutter code samples that can '
+        'be created with --sample to a JSON file.',
```
Comment (pq) [inline packages/flutter_tools/lib/src/commands/create.dart]: Maybe:

    'Specifies a JSON output file for a listing of Flutter code samples that can created with --sample.'

?

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (DanTup) [reply packages/flutter_tools/lib/src/commands/create.dart]: Done!

**Comment by DanTup** [comment]
Thanks! I'll land when it's all green.

@pq I guess we'll have to gate this on the version (or run it and detect the failure). I may remove my url-access code, and just update the locally cached version, and use that in the case this switch isn't available (which I'll probably gate based on the version number, using the next version tagged).

**Comment by DanTup** [comment]
Failures seem unrelated to this:

```
18:20 +608 ~3 -1: test/commands/test_test.dart: flutter test should not have extraneous error messages [E]                                                                                             
  TimeoutException after 0:01:00.000000: Test timed out after 1 minutes.
  package:test_api  Invoker._onRun.<fn>.<fn>.<fn>
```

I'll restart them..

**Comment by pq** [comment]
> @pq I guess we'll have to gate this on the version (or run it and detect the failure). 

Any additional thoughts on how you'll tackle this in VSCode?  I'm inclined to do something essentially like:

> I may remove my url-access code, and just update the locally cached version, and use that in the case this switch isn't available (which I'll probably gate based on the version number, using the next version tagged).

**Comment by DanTup** [comment]
Yeah, I think I'll use the version number. We already read the version as part of the SDK detection (https://github.com/Dart-Code/Dart-Code/blob/49dad040eae214923252d5e7e700cbf04418e116/src/sdk/utils.ts#L222) so I'll just add it to our [Capabilities class](https://github.com/Dart-Code/Dart-Code/blob/eeac1b49bc40fe6d8c43f0bf22405d21419fc482/src/flutter/capabilities.ts) and fall back to the local index if it's too low.

**Pull request merged**